topology/pynest/tests/test_spatial_kernels.py

```diff
@@ -0,0 +1,779 @@
+            '''Plot nodes in the network.'''
+
+            # Adjust size of nodes in plot based on number of nodes.
+            nodesize = max(0.01, round(111. / 11 - self._N / 1100.))
+
+            figsize = (8, 6) if self._dimensions == 3 else (6, 6)
+            fig = plt.figure(figsize=figsize)
+            positions = self._positions()
+            connected = self._target_positions()
+            not_connected = set(positions) - set(connected)
+
+            x1 = [pos[0] for pos in not_connected]
+            y1 = [pos[1] for pos in not_connected]
+            x2 = [pos[0] for pos in connected]
+            y2 = [pos[1] for pos in connected]
+
+            if self._dimensions == 2:
+                plt.scatter(x1, y1, s=nodesize, marker='o', color='grey')
+                plt.scatter(x2, y2, s=nodesize, marker='o', color='red')
+            if self._dimensions == 3:
+                ax = fig.add_subplot(111, projection='3d')
+                z1 = [pos[2] for pos in not_connected]
+                z2 = [pos[2] for pos in connected]
+                ax.scatter(x1, y1, z1, s=nodesize, marker='o', color='grey')
+                ax.scatter(x2, y2, z2, s=nodesize, marker='o', color='red')
+
+            plt.xlabel(r'$x$', size=24)
+            plt.ylabel(r'$y$', size=24)
+            plt.xticks(size=16)
+            plt.yticks(size=16)
+            plt.xlim(-0.505, 0.505)
+            plt.ylim(-0.505, 0.505)
+            plt.subplots_adjust(bottom=0.15, left=0.17)
+
+        def show_CDF(self):
+            '''
+            Plot the cumulative distribution function (CDF) of
+            source-target distances.
+            '''
+
+            plt.figure()
+            x = np.linspace(0, self._max_dist, 1000)
+            cdf = self._cdf(x)
+            plt.plot(x, cdf, '-', color='black', linewidth=3,
+                     label='Theory', zorder=1)
+            y = [(i + 1.) / len(self._target_dists)
+                 for i in range(len(self._target_dists))]
+            plt.step([0.0] + self._target_dists, [0.0] + y, color='red',
+                     linewidth=1, label='Empirical', zorder=2)
+            plt.ylim(0, 1)
+
+            plt.xlabel('Distance')
+            plt.ylabel('CDF')
+            plt.legend(loc='center right')
+
+        def show_PDF(self, bins=100):
+            '''
+            Plot the probability density function of source-target distances.
+
+            Parameters
+            ----------
+                bins: Number of histogram bins for PDF plot.
+            '''
+
+            plt.figure()
+            x = np.linspace(0, self._max_dist, 1000)
+            area = scipy.integrate.quad(self._pdf, 0, self._max_dist)[0]
+            y = np.array([self._pdf(D) for D in x]) / area
+            plt.plot(x, y, color='black', linewidth=3, label='Theory',
+                     zorder=1)
+            plt.hist(self._target_dists, bins=bins, histtype='step',
+                     linewidth=1, density=True, color='red',
```

Comment:
the argument density works with matplotlib 2.2 but for older version it would give an error. Older versions has this argument but it called normed.

Reply:
Ok, reverted to `normed` for backward compatibility.